homeassistant/components/flux_led/light.py

```diff
@@ -171,133 +141,231 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 
 async def async_setup_entry(hass, entry, async_add_entities):
     """Set up the Flux lights."""
-    config_auto = entry.options["global"].get(
-        CONF_AUTOMATIC_ADD, entry.data[CONF_AUTOMATIC_ADD]
-    )
-    config_devices = entry.data[CONF_DEVICES]
-    config_options = entry.options
-
-    lights = []
-
-    if config_auto:
-        # Find the bulbs on the LAN
-        scanner = BulbScanner()
-        await hass.async_add_executor_job(scanner.scan)
-
-        for device in scanner.getBulbInfo():
-            device_id = device["ipaddr"].replace(".", "_")
-            if device_id not in config_devices:
-                config_devices[device_id] = device
-
-    for device_id, device in config_devices.items():
-        add_device = {}
-        add_device["name"] = device.get("name", device_id)
-        add_device[CONF_HOST] = device[CONF_HOST]
-        add_device[CONF_PROTOCOL] = None
-        add_device[ATTR_MODE] = None
-        add_device[CONF_CUSTOM_EFFECT] = None
-        add_device[CONF_EFFECT_SPEED] = config_options.get(device_id, {}).get(
-            CONF_EFFECT_SPEED,
-            config_options.get("global", {}).get(
-                CONF_EFFECT_SPEED, DEFAULT_EFFECT_SPEED
-            ),
+
+    async def async_new_lights(bulbs: dict):
+        """Add new bulbs when they are found or configured."""
+        bulb_coordinator = (
+            hass.data[DOMAIN].get(entry.entry_id, {}).get(BULB_COORDINATOR)
         )
+        lights = []
+
+        for bulb_id, bulb_details in bulbs.items():
+            effect_speed = entry.options.get(bulb_id, {}).get(
+                CONF_EFFECT_SPEED,
+                entry.options.get("global", {}).get(
+                    CONF_EFFECT_SPEED, DEFAULT_EFFECT_SPEED
+                ),
+            )
+
+            coordinator = FluxLEDCoordinator(
+                hass=hass,
+                name=bulb_details[CONF_NAME],
+                update_interval=DEFAULT_SCAN_INTERVAL,
```

Comment:
We don't allow shorter scan interval than 5 seconds.

Reply:
OK - set to 5 seconds